### ISSUE TITLE
createSession directly sets the user as Authenticated

### ISSUE THREAD

**Issue by JaviBonilla** [lohanidamodar/flappwrite_account_kit#2]
This can be the expected behavior in some cases but in others it is required that the user's email in verified, in such cases the `createSession` method must be called to check the `emailVerification` property.

The `createSession` method could be called in these cases:

- Login (check email is verified).
- Singup (send verification email).
- Resend verification email.

I don't know what it is the best way to tackle this, but I suggest to consider these options:

- Have two different AuthStatus states: `authenticatedVerified` and `authenticatedNotVerified` instead of `authenticated`.
- Include and optional parameter `notifyListeners = true` in `createSession` and `deleteSession`, so these methods can be used for signing up and resending verification emails.

**Comment by eldadfux** [comment]
This is the expected behavior. We designed the Account API this way to make sure each CRUD operation is decoupled from the other to avoid extra complexity and limited flexibility using different auth flows in different use-cases.

You can use the `emailVerification` attribute in the [user model](https://appwrite.io/docs/models/user) to introduce a different UI to the user on true/false values, forcing him to verify his account before allowing him to interact with your app, or just to show a notification if thats not a must have requirement for the user. We've seen some  use cases where verification is only required after X days for example.

We have been playing with the idea of introducing a new permission called `user:[USER_ID]/verified` to allow more control over server resources (collections, documents, files, etc...). We would love to get your feedback on that.

**Comment by JaviBonilla** [comment]
Thanks @eldadfux, so for the example in the readme, would it be something like this? Is the `widget` variable rebuilt each time the `authNotifier` changes for the authenticated case? I cannot think of a different way of doing this using the provider, but probably there is a better way to do this.

```dart
class MainScreen extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    final authNotifier = context.authNotifier;

    Widget widget;
    switch (authNotifier?.status ?? AuthStatus.uninitialized) {
      case AuthStatus.authenticated:
        widget = authNotifier!.user!.emailVerification ? AdminPage() : LoginPage();
        break;
      case AuthStatus.unauthenticated:
      case AuthStatus.authenticating:
        widget = LoginPage();
        break;    
      case AuthStatus.uninitialized:
      default:
        widget = LoadingPage();
        break;
    }
    return widget;
  }
}
```

PD: To clarify, I was proposing these changes for this Flutter package (not for AppWrite) since this is a package that makes it easy to use manage authentication and account features in Flutter and using providers.

**Comment by lohanidamodar** [comment]
> Thanks @eldadfux, so for the example in the readme, would it be something like this? Is the `widget` variable rebuilt each time the `authNotifier` changes for the authenticated case? I cannot think of a different way of doing this using the provider, but probably there is a better way to do this.
> 
> ```dart
> class MainScreen extends StatelessWidget {
>   @override
>   Widget build(BuildContext context) {
>     final authNotifier = context.authNotifier;
> 
>     Widget widget;
>     switch (authNotifier?.status ?? AuthStatus.uninitialized) {
>       case AuthStatus.authenticated:
>         widget = authNotifier!.user!.emailVerification ? AdminPage() : LoginPage();
>         break;
>       case AuthStatus.unauthenticated:
>       case AuthStatus.authenticating:
>         widget = LoginPage();
>         break;    
>       case AuthStatus.uninitialized:
>       default:
>         widget = LoadingPage();
>         break;
>     }
>     return widget;
>   }
> }
> ```
> 
> PD: To clarify, I was proposing these changes for this Flutter package (not for AppWrite) since this is a package that makes it easy to use manage authentication and account features in Flutter and using providers.

Thank you for your interest and contribution to the library.

We can discuss more on this, however authentication status is not related to Email verification. so What you did in readMe is actually better approach. After the user is authenticated, you can check whether or not the email is verified for the user. and may be if email is not verified you want to send to another page called `EmailVerification` where user can tap to resend verification email, or update their email entirely.. it's for applications logic to decide.

Let me know what you think.

**Comment by JaviBonilla** [comment]
Hi @lohanidamodar,

It's been a while since we talked about this in Discord. I've proposed some minor changes (PR #5) that would give more flexibility.

- **Create session** with an optional argument to *notify listeners*.
- **Create user** with optional arguments to *notify listeners* and *create session*.

As an example, with these changes the **flappwrite_account_kit** package can be used together the [flutter login package](https://github.com/NearHuscarl/flutter_login) to not login when the user email is not verify. This behavior is optional and not enabled be default.

Let me know what do you think about these changes.

https://user-images.githubusercontent.com/20316179/135713048-1b659413-9327-44b3-911e-a75e0401f7df.mp4

**Comment by lohanidamodar** [comment]
Great work @JaviBonilla. I've merged the changes. Fixed by #5
